server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -780,7 +780,11 @@ protected ExceptionThrowingChannel(RestRequest request, boolean detailedErrorsEn
 
         @Override
         public void sendResponse(RestResponse response) {
-            throw new IllegalStateException("always throwing an exception for testing");
+            try {
+                throw new IllegalStateException("always throwing an exception for testing");
+            } finally {
+                releaseOutputBuffer();
```

Comment:
Meh, best short term fix I could think of, normally this would be taken care of by the `DefaultRestChannel` itself, but I suppose it's good that tests trip without this change.

Reply:
```suggestion
                // the production implementation in DefaultRestChannel always releases the output buffer, so we must too
                releaseOutputBuffer();
```

Also should we be doing this in `AssertingChannel` a few lines up? Can we add an assertion somewhere that `sendResponse` really does always release the output buffer?

Reply:
It's hard to assert that in general because the release happens async in the production implementation (via the listener on the Netty/Transport thread that actually ships out the request).

It seems like we don't need to do this in the asserting channel as we don't have any error situations there. Lets add it when we need it I'd say, keeps the assertions stricter IMO

Reply:
ah yeah good point about the async release, nvm then